tests/ext-activitylog.test.js

```diff
@@ -70,113 +757,123 @@ test('show/hide logs associated with extension id that is checked/unchecked from
     },
   };
 
-  sendMessage.mockImplementation(() => {
-    return Promise.resolve({ existingLogs: [] });
-  });
+  // sendMessage is called by getExistingLogs method on the 1st time.
+  // sendMessage is called for the 2nd time to clear logs from the background.
+  sendMessage.mockResolvedValueOnce({ existingLogs: logs });
+  sendMessage.mockResolvedValueOnce();
 
   document.body.innerHTML = activityLogBody;
 
-  const { activityLog } = new ActivityLog();
-  const extFilterBtn = activityLog.view.extFilter.shadowRoot.querySelector(
-    '.toggle-btn'
-  );
+  const emptyTableLabel = document
+    .querySelector('log-view')
+    .shadowRoot.querySelector('.table-empty-label');
 
-  activityLog.handleNewLogs(logs);
+  const emptyTableLabelHidden = observeChange(emptyTableLabel);
+  const { activityLog } = new ActivityLog();
+  await emptyTableLabelHidden;
 
+  const clearLogBtn = activityLog.view.clearLogBtn;
   const tableBody = activityLog.view.logView.shadowRoot.querySelector('tbody');
   const getTableRows = () => tableBody.querySelectorAll('tr');
-  let tableRows = getTableRows();
 
-  // Initially filter checkbox list is hidden
-  expect(activityLog.view.extFilter.classList.contains('expanded')).toBeFalsy();
+  // expecting 2 rows appended in the table
+  expect(getTableRows().length).toBe(2);
 
-  const extFilterBtnChanged = observeChange(activityLog.view.extFilter);
-  extFilterBtn.click();
-  await extFilterBtnChanged;
+  const [row] = getTableRows();
 
-  // the filter checkbox list dropdown is displayed
-  expect(
-    activityLog.view.extFilter.classList.contains('expanded')
-  ).toBeTruthy();
+  const { logDetailWrapper } = activityLog.view.logView;
+  // Initially the log details is hidden (when no table is clicked)
+  expect(logDetailWrapper.hidden).toBe(true);
+  // clicking a row in the table to show logs details
+  row.click();
+  expect(logDetailWrapper.hidden).toBe(false);
+
+  clearLogBtn.click();
+  await observeChange(logDetailWrapper);
```

Comment:
I see that elsewhere observeChange was called before triggering the change to the DOM elements and its promise returned value stored to be awaited on after triggering the side effect (e.g. lines 771-773 in this same file diff), is there a risk for this one to become an intermittent? was the one 771-773 failing intermittently if we would do the same there?

Reply:
@rpl Nice catch! It felt a bit weird to see this code at line no. 792. I think the code at line no. 792 should follow the same way of resolving promises once the DOM changes like the one at line 771-773 i.e.
- Setup Promise with current state of DOM
- Trigger DOM changes
- Wait for the promise to resolve

Currently, it's not failing because when the `clearLogBtn.click()` event occurs, it does some work i.e. remove logs from background (includes some promise resolving), then remove table rows from DOM. It may fail intermittently if the `clearLogBtn.click()` event had to do very less work. 

> was the one 771-773 failing intermittently if we would do the same there?

No, it's not failing since the `ActivityLog()` have to do some work i.e. fetch all the logs from background (includes some promise resolving) before making changes in the DOM.